openjdk/jdk
# JDK-8316696: Remove the testing base classes: IntlTest and CollatorTest

Please review this PR which removes the i18n related testing base classes `IntlTest` and `CollatorTest` and converts all the tests that use them,

IntlTest and CollatorTest are testing classes which are extended by tests in `text/`, `util/Locale`, `util/TimeZone`, and `util/Calendar`. The abstract testing classes are quite dated and have caused issues such as: variation between OS, hiding stack trace, and causing tests to spuriously pass.

This change mainly automates a low level conversion of all the tests (75) using the frameworks; all tests were converted to use JUnit instead. (With the exception of `DateFormatRoundTripTest` due to the nature of the test).

The main changes can be viewed in the following commits

scripted changes - [c0ece01 ](https://github.com/openjdk/jdk/commit/c0ece01e91479a020d5c6dce937dc827472b763b)
- Converts the IntlTest methods logln, log, err, and errln
- Adds the JUnit Test annotation to tests that should be ran
- Adjusts the Jtreg tags accordingly
- Remove the main method
- Insert initAll() methods for tests that previously adjusted the JVM default Locale/TimeZone in the main method

manual changes - [9a54910](https://github.com/openjdk/jdk/commit/9a5491065a94a4dc7a05194f3b8330efba8077b7)
- Some tests that had extensive logic in the main method or did not follow the general IntlTest format had to be manually adjusted
- Also clarified some tests that had optional argument setup in the main method (now removed)

removal of IntlTest and CollatorTest - [8ee9f9c](https://github.com/openjdk/jdk/commit/8ee9f9c79f79210ee6244186970d87a1cac05556) [f90266f](https://github.com/openjdk/jdk/pull/15954/commits/f90266f4f019c031c5f985dd658f62a02cc8b422) [3c67679](https://github.com/openjdk/jdk/pull/15954/commits/3c676794dd0c703db066a346c36e213d113d9acc)
- Removes IntlTest in both the testlib and under TimeZone/
- Replaces CollatorTest with CollatorTestUtils

Tiers 1-3 clean


---------
### Progress

### Issue
 * [JDK-8316696](https://bugs.openjdk.org/browse/JDK-8316696): Remove the testing base classes: IntlTest and CollatorTest (**Bug** - P4)


### Reviewers
 * [Naoto Sato](https://openjdk.org/census#naoto) (@naotoj - **Reviewer**) ⚠️ Review applies to [47532e8a](https://git.openjdk.org/jdk/pull/15954/files/47532e8a6afe9314ff2370f3620b3fa23ce1135d)


### Reviewing
<details><summary>Using <code>git</code></summary>

Checkout this PR locally: \
`$ git fetch https://git.openjdk.org/jdk.git pull/15954/head:pull/15954` \
`$ git checkout pull/15954`

Update a local copy of the PR: \
`$ git checkout pull/15954` \
`$ git pull https://git.openjdk.org/jdk.git pull/15954/head`

</details>
<details><summary>Using Skara CLI tools</summary>

Checkout this PR locally: \
`$ git pr checkout 15954`

View PR using the GUI difftool: \
`$ git pr show -t 15954`

</details>
<details><summary>Using diff file</summary>

Download this PR as a diff file: \
<a href="https://git.openjdk.org/jdk/pull/15954.diff">https://git.openjdk.org/jdk/pull/15954.diff</a>

</details>


### Webrev
[Link to Webrev Comment](https://git.openjdk.org/jdk/pull/15954#issuecomment-1738303829)
